src/vm/eventpipe.cpp

```diff
@@ -613,30 +648,13 @@ void EventPipe::WriteEventInternal(EventPipeEvent &event, EventPipeEventPayload
     // If the activity id isn't specified AND we are in a managed thread, pull it from the current thread.
     // If pThread is NULL (we aren't in writing from a managed thread) then pActivityId can be NULL
     if (pActivityId == NULL && pThread != NULL)
-    {
         pActivityId = pThread->GetActivityId();
-    }
 
-    for (EventPipeSessions::Iterator iterator = s_pSessions->Begin();
-         iterator != s_pSessions->End();
-         ++iterator)
+    if (s_pRundownSession.LoadWithoutBarrier() != nullptr)
```

Comment:
I think the plan was to make this a thread-static (ie a variable in EventPipeThread) rather than a global. Doing it as a global means that multiple sessions can't do rundown at the same time. This is already currently the case, but the goal is to get rid of that restriction.

As written here when one session does rundown, all threads will stop logging to any other session because satisfying this if means they won't go into the else branch below.

Reply:
I'll change it.